### DISCUSSION TITLE
Explore: add ad hoc filter of parsed fields to log details

### DISCUSSION THREAD

**Discussion by ventris** [grafana/grafana#42313] [Observability]
### What Grafana version are you using?
grafana:master docker image
### What datasource are you using?
Loki
### What OS are you running grafana on?
Debian
### Feature
If ones logs are stored with key/values you get some cool things to play with in the Explore UI, such as click on the key/value and it will give you a dropdown with the key and its most used values. Seen below.
![](https://i.imgur.com/swRey5d.png)

It would be really cool if you could with one click add this key/value that you have selected to your search somehow without having to type it out. This could also be useful for adding one word from the logs to the search and not only key/values.

**Comment by apollo13** [comment]
@ventris Do I need to add something special so grafana will underline the detected fields like in your screenshot? My log lines are in logfmt, but I do not get the shown underline.

**Comment by megamorf** [comment]
@apollo13 You can see those fields in the `Detected fields` section of an event:

![grafik](https://user-images.githubusercontent.com/1732128/122679783-7deb6d00-d1ec-11eb-9417-3a07ab4506fe.png)

**Comment by apollo13** [comment]
Ah, so the initial screenshot was for an older version? Thanks.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This has been implemented. Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find  a solution for what you're looking for and provide additional details for the feature.
